maggy/ablation/ablationstudy.py

```diff
@@ -203,9 +203,28 @@ def __init__(self):
         self.layers = Layers()
         self.base_model_generator = None
 
+        # the list holding custom model generators. 
+        # One (extra) trial will be generated per each item in the list.
+        self.custom_model_generators = [] 
+
     def set_base_model_generator(self, base_model_generator):
         self.base_model_generator = base_model_generator
-
+    
+    def add_custom_model_generator(self, custom_model_generator, model_identifier):
+        """
+        Add a custom model architecture generator, which will result in a single ablation trial. 
```

Comment:
```suggestion
        Add a custom model architecture generator, which will result in a single ablation trial.

```
insert empty line after short description